File: backend/app/schemas/assistant.py
from pydantic import BaseModel, Field


class AssistantChatRequest(BaseModel):
    question: str = Field(min_length=1)
    repository_id: str | None = None
    conversation_id: str | None = None


class AssistantSource(BaseModel):
    source_type: str
    source_id: str | None = None
    title: str


class AssistantChatResponse(BaseModel):
    answer: str
    sources: list[AssistantSource]
    confidence: float
    conversation_id: str

File: frontend/src/types/assistant.ts
export interface AssistantSource {
  source_type: string;
  source_id: string | null;
  title: string;
}

export interface AssistantChatResponse {
  answer: string;
  sources: AssistantSource[];
  confidence: number;
  conversation_id: string;
}

export interface AssistantMessage {
  role: 'user' | 'assistant';
  content: string;
  sources?: AssistantSource[];
}